azurerm/internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -215,6 +215,8 @@ func dataSourceKubernetesCluster() *schema.Resource {
 							Type:     schema.TypeBool,
 							Computed: true,
 						},
+
+						"upgrade_settings": upgradeSettingsSchema(),
```

Comment:
since this is a Data Source which is read-only this'll need to be:

```suggestion
						"upgrade_settings": upgradeSettingsForDataSourceSchema(),
```